nautobot/extras/models/metadata.py

```diff
@@ -135,3 +141,298 @@ def save(self, *args, **kwargs):
+                #     raise ValidationError(f"Value must be at least {self.metadata_type.validation_minimum}")
+                # if self.metadata_type.validation_maximum is not None and value > self.metadata_type.validation_maximum:
+                #     raise ValidationError(f"Value must not exceed {self.metadata_type.validation_maximum}")
+            # Validate integer
+            elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_FLOAT:
+                try:
+                    value = float(value)
+                except ValueError:
+                    raise ValidationError("Value must be a float.")
+            # Validate boolean
+            elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_BOOLEAN:
+                try:
+                    value = is_truthy(value)
+                except ValueError as exc:
+                    raise ValidationError("Value must be true or false.") from exc
+
+            # Validate date
+            elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_DATE:
+                if not isinstance(value, date):
+                    try:
+                        datetime.strptime(value, "%Y-%m-%d")
+                    except TypeError:
+                        raise ValidationError("Value must be a date or str object.")
+                    except ValueError:
+                        raise ValidationError("Date values must be in the format YYYY-MM-DD.")
+            # Validate datetime
+            elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_DATETIME:
+                acceptable_datetime_formats = [
+                    "YYYY-MM-DDTHH:MM:SS",
+                    "YYYY-MM-DDTHH:MM:SS(+,-)zzzz",
+                    "YYYY-MM-DDTHH:MM:SS(+,-)zz:zz",
+                ]
+                if not isinstance(value, datetime):
+                    try:
+                        datetime.strptime(value, "%Y-%m-%dT%H:%M:%S%z")
+                    except ValueError:
+                        try:
+                            # No time zone info entered so assume UTC timezone
+                            datetime.strptime(value, "%Y-%m-%dT%H:%M:%S")
+                            value += "+0000"
+                        except ValueError:
+                            raise ValidationError(
+                                f"Datetime values must be in the following formats {acceptable_datetime_formats}"
+                            )
+                    except TypeError:
+                        raise ValidationError("Value must be a datetime or str object.")
+                else:
+                    # if value is a datetime object
+                    # check if datetime object has tzinfo
+                    if value.tzinfo is None:
+                        value = value.replace(tzinfo=timezone.utc)
+                    value = value.replace(microsecond=0).isoformat()
```

Comment:
Why's the timezone conversion needed? Does `isoformat()` not already account for time zones?